### ISSUE TITLE
Install flash-attention error：ERROR: Could not build wheels for flash-attn, which is required to install pyproject.toml-based projects

### ISSUE THREAD

**Issue by zyh3826** [Dao-AILab/flash-attention#601]
system：ubuntu
cuda：12.2
gpu：A800
cmd：`MAX_JOBS=4 python -m pip install flash-attn --no-build-isolation`
error msg:
![image](https://github.com/Dao-AILab/flash-attention/assets/31238754/fe721a1c-9fb7-478b-bb04-a9561f7dcb60)
this is my first error.

I noticed that there has a URL `https://github.com/Dao-AILab/flash-attention/releases/download/v2.3.2/flash_attn-2.3.2+cu118torch2.2cxx11abiFALSE-cp310-cp310-linux_x86_64.whl`, I just download this wheel and install it using `pip`, it worked, but when I using `import flash_attn_2_cuda as flash_attn_cuda`， another error occurred：
![image](https://github.com/Dao-AILab/flash-attention/assets/31238754/1a068c86-abbd-4577-b93b-133c8d28e55f)

hope you can help me solve these two errors， thanks a lot.

**Comment by tridao** [comment]
I don't think we have wheels for pytorch 2.2 yet.

**Comment by zyh3826** [comment]
> I don't think we have wheels for pytorch 2.2 yet.

yes, I made a mistake, I downloaded the `2.0.1` version, but I upgraded torch to 2.2 just a few hours ago. Now I'm trying compile it from source, thank your reply

**Comment by zyh3826** [comment]
@tridao I compile it from the source, and everything is fine, but when I `import` it I get this error:
![image](https://github.com/Dao-AILab/flash-attention/assets/31238754/f325ca8d-9288-4807-ab12-be585fbfc691)

**Comment by tridao** [comment]
Sounds like some incompatibly in pytorch version. Can you try with pytorch 2.1?

**Comment by zyh3826** [comment]
> Sounds like some incompatibly in pytorch version. Can you try with pytorch 2.1?

`pytorch 2.1` is fine, thanks

**Comment by Batwho** [comment]
@zyh3826 Hi, I'm facing a quite similar bug as yours. I notice that you are also trying to apply flash_attn on LLaMa. Could you please share your more about how you solve the issue? In particular, I also tried compiling from source, but the gcc compiler is never right and fail on "timespec_get". If you happen to encounter this issue as well, how do you solve that?

**Comment by zyh3826** [comment]
> @zyh3826 Hi, I'm facing a quite similar bug as yours. I notice that you are also trying to apply flash_attn on LLaMa. Could you please share your more about how you solve the issue? In particular, I also tried compiling from source, but the gcc compiler is never right and fail on "timespec_get". If you happen to encounter this issue as well, how do you solve that?

@Batwho sorry for reply late，I install this package by using wheel, downloading at [this page](https://github.com/Dao-AILab/flash-attention/releases)

**Comment by jc-louis** [comment]
The issue seems fixed
- ❌ torch 2.2.0 + flash-attn 2.5.0
- ✅ torch 2.2.1 + flash-attn 2.5.5
